packages/vitest/src/runtime/collect.ts

```diff
@@ -97,11 +104,16 @@ function interpretTaskModes(suite: Suite, namePattern?: string | RegExp, onlyMod
     else if (t.type === 'suite') {
       if (t.mode === 'skip')
         skipAllTasks(t)
+      else
+        interpretTaskModes(t, namePattern, onlyMode)
 
       // if all subtasks are skipped, marked as skip
       if (t.mode === 'run') {
-        if (t.tasks.every(i => i.mode !== 'run'))
+        if (t.tasks.every(i => i.mode !== 'run')) {
           t.mode = 'skip'
+          if (t.file)
+            t.file.mode = 'skip'
+        }
```

Comment:
If we have a nested suite where all subtasks are skipped, this check could mark a whole file as skip when there are tasks to run in other suites in the file. Maybe we could add an extra check out in line 74 to mark the file as skip, or refactor it out of the loop and add it at the end of interpretTaskMode

Reply:
Ah yes, totally! I'll take another look

Reply:
I pushed an update.

You were right, too many test files were skipped.
The workaround requires a check for (transitive) parent suites that are marked as `only`.